File: VB/UnboundListView.Module/Module.vb
Imports System
Imports System.Collections.Generic

Imports DevExpress.ExpressApp
Imports System.Reflection


Namespace UnboundListView.Module
    Public NotInheritable Partial Class UnboundListViewModule
        Inherits ModuleBase

        Public Sub New()
            InitializeComponent()
        End Sub
        Public Overrides Sub CustomizeTypesInfo(ByVal typesInfo As DevExpress.ExpressApp.DC.ITypesInfo)
            MyBase.CustomizeTypesInfo(typesInfo)
            Dim typeInfo = TryCast(typesInfo.FindTypeInfo(GetType(Duplicate)), DevExpress.ExpressApp.DC.TypeInfo)
            typeInfo.AddKeyMember(typeInfo.FindMember("Id"))
        End Sub
    End Class
End Namespace

File: CS/UnboundListView.Module/Module.cs
using System;
using System.Collections.Generic;

using DevExpress.ExpressApp;
using System.Reflection;


namespace UnboundListView.Module {
    public sealed partial class UnboundListViewModule : ModuleBase {
        public UnboundListViewModule() {
            InitializeComponent();
        }
        public override void CustomizeTypesInfo(DevExpress.ExpressApp.DC.ITypesInfo typesInfo) {
            base.CustomizeTypesInfo(typesInfo);
            var typeInfo = typesInfo.FindTypeInfo(typeof(Duplicate)) as DevExpress.ExpressApp.DC.TypeInfo;
            typeInfo.AddKeyMember(typeInfo.FindMember("Id"));
        }
    }
}